### PR TITLE
ABC controller test

### PR DESCRIPTION
**Pull request by MichaelClerx**
Fixes several oddities in ABC controller test:

- A file called "True" was being written on every run
- Logged output was not compared to expected results
- Logging tests used 1000s of iterations, making the logging test take 7 seconds on my machine
- A new ABCController was created but not used for most of the logging tests

### CODE REVIEW

**Comment by MichaelClerx** [comment]
Anyone feel like reviewing this one? Fixes some annoyances in a test, no hard maths or anything :D @pints-team/devs ?

File: pints/tests/test_abc_controller.py [hunk 1]
```diff
@@ -115,58 +156,125 @@ def test_logging(self):
...
 
         # With output to screen
         np.random.seed(1)
         with StreamCapture() as capture:
-            pints.ABCController(
+            abc = pints.ABCController(
                 self.error_measure, self.log_prior, method=pints.RejectionABC)
             abc.set_max_iterations(10)
+            abc.set_n_samples(3)
             abc.set_log_to_screen(True)
```
Comment (DavAug) [inline pints/tests/test_abc_controller.py:178]: Could move lines 174-178 also outside the 'with' statement.

File: pints/tests/test_abc_controller.py [hunk 2]
```diff
@@ -115,58 +156,125 @@ def test_logging(self):
...
-            pints.ABCController(
+            abc = pints.ABCController(
                 self.error_measure, self.log_prior, method=pints.RejectionABC)
-            abc.set_max_iterations(10)
-            abc.set_log_to_screen(False)
-            abc.set_log_to_file(True)
+            abc.set_max_iterations(6)
+            abc.set_log_interval(6, 3)
+            abc.set_parallel(2)
+            abc.set_log_to_screen(True)
```
Comment (DavAug) [inline pints/tests/test_abc_controller.py:199]: Same here

File: pints/tests/test_abc_controller.py [hunk 3]
```diff
@@ -115,58 +156,125 @@ def test_logging(self):
...
+        with TemporaryDirectory() as d:
+            filename = d.path('test.txt')
+            abc = pints.ABCController(
+                self.error_measure,
+                self.log_prior,
+                method=pints.RejectionABC)
+            abc.set_max_iterations(6)
+            abc.set_log_interval(6, 3)
+            abc.set_log_to_screen(False)
+            abc.set_log_to_file(filename, csv=True)
```
Comment (DavAug) [inline pints/tests/test_abc_controller.py:253]: I think you could also move some lines out of the with statement

**Review by DavAug — APPROVED** [review]
Thanks @MichaelClerx ! I have some minor comments that you can happily ignore 😊

**Pull request merged**